### DISCUSSION TITLE
processor_add_fields添加属性 插件 不能使用processor_regex 解析出的变量值，是否可以增强该功能

### DISCUSSION THREAD

**Discussion by BruceWayneVsSuperMan** [alibaba/ilogtail#948] [Help]
需求场景：
使用ilogtail采集springboot日志时，使用processor_regex解析日志后无法在processor_add_fields插件中使用解析出的变量值
如下代码，我想添加一个string_valuses 的字段，并用日志信息中的某些字段组成数组为其赋值
`processors:
  - Type: processor_add_fields
    Fields:
      project: sso
  - Type: processor_regex
    SourceKey: content
    Regex: '\[(?P<datetime>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}\.\d+)\] \[(?P<node>[^\]]+)\] \[\] \[Id\(\d+\):(?P<thread>[^\]]+)\] \[(?P<level>[A-Z]+)\] \[(?P<class>[^\]]+):\d+\s*\] - (?P<message>.+)'
    Keys:
      - datetime
      - node
      - thread
      - level
      - class
      - message
    NoKeyError: true
    NoMatchError: true
  - Type: processor_add_fields
    Fields:
      string_values: [$node , $thread ,$message]`

**Comment by messixukejia** [comment]
目前不支持这个功能，是否可以一起来开发下？
